src/python/dxpy/scripts/dx.py

```diff
@@ -2513,22 +2513,16 @@ def install(args):
 
 def uninstall(args):
     app_desc = get_app_from_path(args.app)
-    if app_desc is None:
-        try:
-            dxpy.api.app_uninstall(args.app)
-            print('Uninstalled application with id: ' + args.app)
-        except:
-            try:
-                dxpy.api.app_uninstall('app-' + args.app)
-                print('Uninstalled application with name: ' + args.app)
-            except:
-                err_exit()
+    if app_desc:
+        try_call(dxpy.api.app_uninstall, app_desc['id'])
+        print('Uninstalled the {name} app'.format(name=app_desc['name']))
     else:
-        try:
-            dxpy.api.app_uninstall(app_desc['id'])
-            print('Uninstalled the ' + app_desc['name'] + ' app')
-        except:
-            err_exit()
+        user_data = dxpy.api.user_describe(dxpy.whoami(), {"fields": {"appsInstalled": True}})
+        if args.app in user_data['appsInstalled']:
+            try_call(dxpy.api.app_uninstall, 'app-' + args.app)
+        else:
```

Comment:
Let me show an example. Let the user be user-pnguyen.

```
pnguyen@pnguyen:~/env$ dx api user-`dx whoami` describe '{"fields": {"appsInstalled": true}}'
{
    "appsInstalled": {
        "url_fetcher": true
    }, 
    "id": "user-pnguyen"
}
```
For sake of example, suppose that user-pnguyen no longer has any authorizations to `url_fetcher`. I think it will make sense for the following to occur:

- `dx uninstall url_fetcher`: Uninstalls `url_fetcher` from the profile of user-pnguyen.
- `dx uninstall app-url_fetcher`: Uninstalls `url_fetcher` from the profile of user-pnguyen.
- `dx uninstall app-url_fletcher`: (**Note the typo**.) Throws an error. This makes it more intuitive to user-pnguyen that he was not able to uninstall the app that he may have wanted to uninstall. Does this make sense?

Additionally, we want to preserve the ability to specify the app by its path (hence the need to maintain `app_desc = get_app_from_path(args.app)`).